nca/Parsers/IstioPolicyYamlParser.py

```diff
@@ -485,30 +489,41 @@ def parse_ingress_rule(self, rule):
         to_array = self.get_key_array_and_validate_not_empty(rule, 'to')
         # currently parsing only ports
         # TODO: extend operations parsing to include other attributes
+        conn_props = ConnectivityProperties.make_empty_props()
         if to_array is not None:
             connections = ConnectionSet()
             for operation_dict in to_array:
-                connections |= self.parse_operation(operation_dict)
+                conns = self.parse_operation(operation_dict)
+                connections |= conns
+                conn_props |= conns.convert_to_connectivity_properties(self.peer_container)
         else:  # no 'to' in the rule => all connections allowed
             connections = ConnectionSet(True)
+            conn_props = ConnectivityProperties.make_all_props()
 
         # condition possible result value:
         #         source-ip (from) , source-namespace (from) [Peerset], destination.port (to) [ConnectionSet]
         # should update either res_pods or connections according to the condition
         condition_array = rule.get('when')  # this array can be empty (unlike 'to' and 'from')
         # the combined condition ("AND" of all conditions) should be applied
+        condition_conns = ConnectionSet(True)
         if condition_array is not None:
             for condition in condition_array:
                 condition_res = self.parse_condition(condition)
                 if isinstance(condition_res, PeerSet):
                     res_peers &= condition_res
                 elif isinstance(condition_res, ConnectionSet):
-                    connections &= condition_res
-
+                    condition_conns &= condition_res
         if not res_peers:
             self.warning('Rule selects no pods', rule)
-
-        return IstioPolicyRule(res_peers, connections)
+        condition_props = condition_conns.convert_to_connectivity_properties(self.peer_container)
```

Comment:
same `ConnectionSet` dependency issue here

Reply:
Removed the dependency